### PR TITLE
connect "foreach"

### PR DESCRIPTION
**Pull request by JeremyMonarch**
Add possibility use "foreach" with my custom stack

### CODE REVIEW

File: Preparation/MyStack/MyStackInter.cs [hunk 1]
```diff
@@ -33,27 +33,39 @@ public bool IsEmpty()
...
             {
-                Console.WriteLine("Stack is null");
+                throw new InvalidOperationException("Stack is empty!");
             }
             Console.WriteLine("Your stack is");
             for (int i = _count; i >= 0; i--)
             {
                 Console.WriteLine(_underliyngArray[i]);
             }
         }
```
Comment (Nazakato) [inline Preparation/MyStack/MyStackInter.cs]: I don;t think this methods is needed here anymore...

File: Preparation/MyStack/MyStackInter.cs [hunk 2]
```diff
@@ -33,27 +33,39 @@ public bool IsEmpty()
...
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            for (int i = 0; i < _underliyngArray.Length; i++)
+            {
+                yield return _underliyngArray[i];
+            }
```
Comment (Nazakato) [inline Preparation/MyStack/MyStackInter.cs]: Do you know haw exactly `yield return` works under the hood? That is a possible an simple solution, "syntax sugar", but please return an instance of your class that will implement `IEnumerator<T>`. Something like `public class MyStackEnumerator<T> : IEnumerator<T>`

File: Preparation/MyStack/MyStackInter.cs (hunk 1 above)
Comment (JeremyMonarch) [reply Preparation/MyStack/MyStackInter.cs]: Which method? IsEmpty or PrintSack?

File: Preparation/MyStack/MyStackInter.cs (hunk 2 above)
Comment (JeremyMonarch) [reply Preparation/MyStack/MyStackInter.cs]: Yes, i know. I investigate code, which leave for me and I understand that it`s custom version of "yield". If it need, I can re-write code with additional class like `MyStackEnumerator<T> : IEnumerator<T>`

Comment (JeremyMonarch) [reply Preparation/MyStack/MyStackInter.cs:63]: If it need, I can re-write code with additional class like `MyStackEnumerator<T> : IEnumerator<T>`

File: Preparation/MyStack/MyStackInter.cs (hunk 1 above)
Comment (JeremyMonarch) [reply Preparation/MyStack/MyStackInter.cs]: Oh, maybe, I know what about you. It will be correct replace `PrintStack` on `foreach`?

Comment (Nazakato) [reply Preparation/MyStack/MyStackInter.cs]: Print stack was needed until you implemented for each support. Now it's unneeded.

File: Preparation/MyStack/MyStackInter.cs (hunk 2 above)
Comment (Nazakato) [reply Preparation/MyStack/MyStackInter.cs]: Yes, do it please :) Basically, it is not 'yield return' a "custom version of enumerator", but when you use yield return, the compiler creates an implementation of enumerator for you. And not the most effecient version in some cases.

**Pull request merged**